### ISSUE TITLE
Design Idea: acquiring water resource

### ISSUE THREAD

**Issue by mokun** [mars-sim/mars-sim#154]
Water is a life support resource and is quintessential for life.

What are the ways of modeling a self-sustaining water supply in future Mars settlements ?

From the point of view of water recycling, if we look at our current state-of-the-art technology, we know NASA has a Regenerative ECLSS Water Recovery System (WRS) onboard the International Space Station (ISS).

WRS comprises two parts: the Urine Processor Assembly (UPA) and Water Processor Assembly (WPA). 

When combined, WRS can recycle waste water such as urine and flush water and into potable water. 

In mars-sim, we have the following 3 resource processes that handles water recycling and production : 
- `Black Water Processor` - recycles black water into grey water. 
- `Grey Water Processor` - recycles grey water into clean water for consumption.
- 'Sabatier RWGS Reactor' - consumes CO2 and H2 and produces CH4, CO and H2O

These systems are not considered long term solution for the expansion of the human settlement.

To deal with the shortage of water, we created the `DigLocalIce` Task and `Collect Ice Resource` Mission for excavating ice.

The settlers may melt the ice and convert it into grey water, regolith and rock salt through a `Manufacturing Process`.

How should we model the presence of Ice on the surface of Mars ? We have assumed that they are easily available.

If ice turns out to be very hard to find locally, it would make sense to look into defining a model of extracting subterranean water for use.

Feel free to share your idea of how we should integrate into mars-sim when it comes to water modeling, in conjunction with what you have read from good literature out there.

**Comment by Urwumpe** [comment]
I am pretty sure, good potable water should be a very rare resource on Mars. But by the latest discoveries, a layer of brine seems to be likely low underground in sedimentary regions, also the amount of ice at the poles is pretty high. So, compared to the moon, its at least not completely dry. 

What is still unknown: How far from the poles does a "martian permafrost" reach? By the areography, it seems to exist, the pictures of the Phoenix lander also suggests that a seasonally freezing and thawing landscape exists on Mars, which again suggests a higher water content in the ground as previously assumed. 

So, maybe it makes sense to have multiple alternative ways to access water. If there is a brine "aquifier", a desalination plant would be needed. For collecting meaningful amounts of water from ice, some sort of automated mining equipment would be needed. 

Another sci-fi idea is of course using ice from asteroids, comets or the rings of Saturn (see "The Expanse"). But getting this done in a scale that replaces the losses of even a small Mars population sounds unfeasible for me. It should be Martian water resources for a long time. 

What about allowing different models of a "Martian hydrology" when creating a simulation? For example, assuming a rather wet or a rather dry Mars? The landing of Insight today will likely provide some important geologic data that could help to refine such models, such as the heat conduction properties of the soil. If we allow customization there, we could easier keep up with the progress in real Mars science. 

I don't know how the German mole probe handles getting submerged in saltwater - this would be a really nice bad luck event in my eyes. I should ask the DLR engineers.

**Comment by mokun** [comment]
> a layer of brine seems to be likely low underground in sedimentary regions, also the amount of ice at the poles is pretty high.

So if a player builds a settlement nowhere near the pole, then we should probability make digging ice harder, unless they are near some crater or chasm, like the way National Geographic's Mars series portrayed ?

>  If there is a brine "aquifier", a desalination plant would be needed. For collecting meaningful amounts of water from ice, some sort of automated mining equipment would be needed.

So we need to create some sort of equipment for the detection and mining and describing ways of reaching into that layer of brine aquifer then, right ?

**Comment by Urwumpe** [comment]
Not sure if there is even ice in equatorial regions. Most examples I know are at higher latitudes. 

Just look at this remote sensing result: 

https://upload.wikimedia.org/wikipedia/commons/b/b0/Water_equivalent_hydrogen_abundance_in_the_lower_latitudes_of_Mars_01.jpg

Looks like Alpha Base would be at a rather bad location, though not the worst. 

We could use this map maybe for creating a near-surface hydrology of Mars. 

And right, some equipment might make sense there.

**Comment by mokun** [comment]
> https://upload.wikimedia.org/wikipedia/commons/b/b0/Water_equivalent_hydrogen_abundance_in_the_lower_latitudes_of_Mars_01.jpg

So this map is from MOLA and it used neutron spectrometer to detect H2 atoms.

On the other hand, Mars Odyssey used gamma ray spectrometer to detect H2 atoms. See https://mars.nasa.gov/odyssey/mission/instruments/grs/

No doubt, the result is similar. 

The verdict is that we will reduce the local occurrence of ice for for equatorial settlements.

Before we can come up with underground aquifer drilling, the temporary solution is to add more `Sabatier RWGS Reactor` to supply water, right ?

**Comment by mokun** [comment]
But at 30 E 0 N, we do have some signature of H2O

**Comment by mokun** [comment]
I read a NASA ISRU Water Study online. 

Based on the data set from two radar instruments (`MARSIS` and `SHARAD`) they pretty much ruled out the notion of the existence of deep underground water between 200m to 300m, where signal is strongest, though they said "the dielectric properties of the Martian subsurface leaves several uncertainties in the radar data interpretation."

See [Water Study](https://mepag.jpl.nasa.gov/reports/Mars_Water_ISRU_Study.pdf) and [MARSIS,  SHARAD](https://www.lpi.usra.edu/meetings/lpsc2007/pdf/1619.pdf)

**Comment by Urwumpe** [comment]
A pretty hard ruling, but well. Actually they rule it out at lower depths than 200-300m, likely because of the SHARAD data, since MARSIS has a 100m resolution. Don't think its really a 100% certainty there, because too little is known about the mars soil properties to be sure (and it was never measured yet), but even a 90% certainty would make this a pretty gamble. So, lets not implement it or shove it to a later for other celestial bodies. So, less water and more ice - bit that far away from the places where we have our initial settlements now. We would need a kind of ice-mining settlement objective maybe and a higher-latitude base... 

But as devils advocate there: If I remember correctly though - reflections by radar are caused by sudden changes in the dielectric properties. A gradual change would not appear. Maybe the soil is too similar to the water that should be detected. Thats my 1% chance. :D

**Comment by mokun** [comment]
In terms of statistics, I'm nearly done with providing the average water usage vs. production in a settlement. 

This is no doubt an important piece of information on the health and well-being of a settlement.

If the net rate is positive, it's good, or else, it's bad.

Since it's an indicator of how well things are running. it should deserve contributing to the overall score of the settlement.

**Comment by Urwumpe** [comment]
Yes it is, like like the oxygen reserves and other ECLSS vitals. 

Can you maybe provide a statistic like "at the current net consumption, the water reserves will run out in x sols" or "At the current net production, grey water will reach its storage capacity in x sols"

**Comment by mokun** [comment]
> oxygen reserves and other ECLSS vitals.

Right, O2 and food as well in near future :D 

For now, it will show :  
 - Water Reserve [kg]
 - Greenhouse operation [kg/sol]
 - Hygiene usage [kg/sol]
 - Resource Processing/Production [kg/sol]

> at the current net consumption, the water reserves will run out in x sols" or "At the current net production, grey water will reach its storage capacity in x sols"

Yay once the basic is coded and no surprise is found, we'll refine this thing along the way.

I'm looking at the difference between **projected** usage vs. **actual** usage.

That's because, we don't really have a detail model of when and how someone would need to relief himself. 

There are many little things we may not have considered in details in terms of water usage and they can be tedious.

There's no actual data for toilet usage per se. 

Perhaps the 20 kg of overall water usage per person per sol seems like a reasonable estimate.

On top of that, we factor is a number of other things such as water ratio level and how many are out on mission, etc.

**Comment by mokun** [comment]
![rwgs](https://user-images.githubusercontent.com/1168584/50311260-66836b80-0459-11e9-8ea5-64ab179dc6ba.jpg)

**Comment by mokun** [comment]
@Urwumpe ,

I found this [youtube video](https://www.youtube.com/watch?v=eSofBhglpCg) showcasing the ability of [JMARS 2035](https://jmars.asu.edu/) to create a zone of hydrated minerals (based on real world data) inside a crater for a future human exploration scenario.

**Comment by shirishag75** [comment]
its in a sense ironic or funny. Spitballing here but I always feel we need to mash or add somehow a planetoid or something which adds and makes gravity more earth-like than it currently is as well as have a heavy atmosphere. But doesn't seem either of them will be a reality for generations to come or probably even considered as we probably don't know if and how orbits may change if mars becomes more heavy or its center of gravity changes.

**Comment by mokun** [comment]
@shirishag75 , if you mean *terraforming* Mars, it's unlikely settlers will be able to achieve that level of impact in mars-sim. We'll have to create another open source project to simulate how the environment of Mars may be transformed globally :D

**Comment by shirishag75** [comment]
yup, meant terraforming mars and yeah, know that's a whole ball-game altogether :(

**Comment by mokun** [comment]
Just added this [wiki](https://github.com/mars-sim/mars-sim/wiki/Mineral-Prospecting-and-Water-Resources). It documents briefly how we model mineral exploration and extraction of water from minerals.

**Comment by mokun** [comment]
I was gonna say that so far we have not included any water maps (i.e. maps showing the possibility of finding water) in mars-sim. 

Although it's still quite speculative if the H2 atoms detected are of any good indication of excavating or extracting water from ores/minerals, I suppose It's about time we add a water map from MOLA and one from GRS as discussed previously in this thread :D

**Comment by mokun** [comment]
@Urwumpe , 

I can't find a good plain Mars water map for our use ? They are either a post-processed pdf file or have x and y axis thrown in

Can you find one ? 

For now, I'm going to **vary** the factitious *ice collection rate* (it used to be set at 20 kg/millisol) in `DigLocalIce` class.

Now, it will vary depending on 2 factors : 
- the elevation 
- the surrounding terrain gradient

I would much prefer to drill for underground water than having the settlers to go outside of the settlement to dig the ice but we will need to conceptual what a *drilling task* would require. And we can discuss it more here.

**Comment by Urwumpe** [comment]
Can we just use a point cloud? Like having a general list of many samples with latitude, longitude and a scalar value, that could for example represent water ice content in low soil?

Maybe we can then ask the scientists behind the PDF maps for the less processed data.

**Comment by mokun** [comment]
> Can we just use a point cloud? Like having a general list of ...

It would be a good idea to visualize things in future.

What do you expect the point cloud will bring ? Any existing Java library you can think of ? 

For now, what we have is just 2-dimensional water data, not 3-D per se., We pretty much assume as if the water ice, if present, can extracted right at the same elevation of a x,y coordinate.

We don't have any *depth* data unless we come up with a model for predicting the presence of water  ice (at our chosen of height resolution) underground.

**Comment by Urwumpe** [comment]
Well, I think about a 2D point cloud there - we could also use a grid, but that would require that the data is globally available.  With a point cloud, we just need the raw data, and can easily interpolate missing swaths of data. I am not convinced that using bitmaps as input is reliable.

For deeper geology, we could use more complex models, but I suspect, since most knowledge of the layers of Mars will depend on drilling cores and layers visible on the surface, a 2D point cloud with a 1D virtual "drill hole" would work better. We would then just need a system to identify layers and interpolate between the virtual drill holes correctly. 

With a point cloud, the big advantage would also be: We could directly store interpolated samples as new data points  to improve performance. And if we fake most of it, we could use drilling missions to create the virtual geology as needed: Initially there are just some random "key points" in the geology maps that define the geology for a simulation run. Later, the missions with some randomness create new geological data points (like the algorithm that splits a triangle to produce mountains, I forgot how it is called).

**Comment by mokun** [comment]
> the layers of Mars will depend on drilling cores and layers visible on the surface, a 2D point cloud with a 1D virtual "drill hole" would work better. We would then just need a system to identify layers and interpolate between the virtual drill holes correctly.

This would be interesting, especially when we are able to visualize in 3D some of the info underground, not just for the water ice but for mineral content as well.

**Comment by mokun** [comment]
I'm reading up on this :
https://en.wikipedia.org/wiki/Point_Cloud_Library

**Comment by Urwumpe** [comment]
Yeah, maybe a simple two view layout would do it for the start: One to show where the hole is located and one to show the layers below.

**Comment by mokun** [comment]
Here's an article called [Fuel and oxygen harvesting from Martian
regolithic brine](https://www.pnas.org/content/pnas/117/50/31685.full.pdf).

We do need to create a modeling of extracting underground water ice. 

Water ice is crucial for creating hydrogen for fuel. 

Obviously, settlers need water.

**Comment by mokun** [comment]
@bevans2000 , @Urwumpe 

What do you see as a quick way of modifying the existing DigLocal class for drilling underground for water ice.

Currently, we send each settler out to dig ice and regolith laboriously. These menial tasks should be for emergency only. 

We should have automated the extraction processes by using machines.

**Comment by Urwumpe** [comment]
What about building a brine well building in the base? This could likely be done by XML alone. If we would have the option to create new bases, we could use prefabs to only build a well and a small power supply for it and let it produce brine automatically. 

Next, we would need some way to process the brine. Also, a pure tanker rover might be useful, if such goods are transported more often.

**Comment by mokun** [comment]
> a brine well building in the base?

ah ic. 

> a pure tanker rover

that's a good idea !

**Comment by mokun** [comment]
@Urwumpe @bevans2000 

How about making use of these two foundations in construction.xml for drilling and excavating water ice?

```
		<foundation name="shallow borehole drilling site" width="5.0" length="10.0" base-level="0"
		 constructable="true" salvagable="true" work-time="10.0" skill-required="3" >
		 	<part name="bore drill bit" number="1" />
		 	<part name="bore drill pipe" number="20" />
			<vehicle type="light utility vehicle" >
				<attachment-part name="crane boom" />
				<attachment-part name="drilling rig" />
			</vehicle>
		</foundation>

		<foundation name="deep borehole drilling site" width="10.0" length="15.0" base-level="0"
		 constructable="true" salvagable="true" work-time="15.0" skill-required="4" >
		 	<part name="bore drill bit" number="4" />
		 	<part name="bore drill pipe" number="60" />
			<vehicle type="light utility vehicle" >
				<attachment-part name="crane boom" />
				<attachment-part name="drilling rig" />
			</vehicle>
		</foundation>
```

**Comment by Urwumpe** [comment]
I think these are at least good starting points. Don't we also have buildings that use this foundation already?

What we then just need is a brine processing plant, unless we directly install it on top of the foundation. Ideally one that can be sent nearly completely as prefab from Earth, so its quickly finished and operating.

Can we also have some tag or flag to such a construction site of the type drilling, that could allow scientists to use the construction/drilling to collect rock or regolith samples? Would be a nice interaction.

**Comment by mokun** [comment]
> Don't we also have buildings that use this foundation already?

Yes, currently, the "deep borehole drilling site" is being used for building the frame of "large areothermal well frame", which in turns for constructing the "large areothermal well".

Similarly, the "deep borehole drilling site"  is for constructing "small areothermal well".

So, it seems to make sense to add a new building called "regolith excavator" or something by building on top of a "deep borehole drilling site".

Similarly, we could enable the building of an "ice excavator" building after having "deep borehole drilling site" in place.

**Comment by mokun** [comment]
I've added this [wiki](https://github.com/mars-sim/mars-sim/wiki/Water-Ice) on the water ice.

Today, I came across this article called [Sublimation of terrestrial permafrost and the implications for ice-loss processes on Mars](https://www.nature.com/articles/s41467-019-09410-8). It discussed about sublimation rate on these precious ice sheet. 

![Images of potential sublimation landforms on Mars](https://media.springernature.com/full/springer-static/image/art%3A10.1038%2Fs41467-019-09410-8/MediaObjects/41467_2019_9410_Fig1_HTML.png?as=webp)

**Comment by mokun** [comment]
> @bevans2000 @Urwumpe 

Once we fully fixed the Construction Mission, shall we create a new **Ice Drilling Mission** ? 

Say we have settlers using an "ice excavator" on "deep borehole drilling site" as suggested early.

**Comment by mokun** [comment]
@bevans2000 @Urwumpe 

> Can we also have some tag or flag to such a construction site of the type drilling, that could allow scientists to use the construction/drilling to collect rock or regolith samples? Would be a nice interaction.

If you mean having a biologist sampling water with possible biosignature... I love that we can create a new task for biologists and chemists :D 

Do you mean the scenario being right in the midst of foundation construction ?

At the very least, we could create a brand new Mission type such as drilling water ice.

**Comment by bevans2000** [comment]
> > @bevans2000 @Urwumpe
> 
> Once we fully fixed the Construction Mission, shall we create a new **Ice Drilling Mission** ?
> 
> Say we have settlers using an "ice excavator" on "deep borehole drilling site" as suggested early.

Yes that seems good.

**Comment by mokun** [comment]
We'll wait for the new Project Mission implementation to come online first. 

Then we'll explore adding new missions such as Tunneling project and Excavation project.
